tools/make-sysext

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+# This file is part of Cockpit.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+
+# Build and install tree as systemd-sysext.
+# See ../HACKING.MD section "Working on your local Fedora machine: systemd-sysext"
+
+set -eu
+
+SYSEXT_NAME="cockpit-git"
+SYSEXT_DIR="/run/extensions/$SYSEXT_NAME"
+
+if [ "${1:-}" = install ]; then
+    if systemd-sysext --no-legend status | grep -qw "$SYSEXT_NAME"; then
+        systemctl stop cockpit.socket
+        systemd-sysext unmerge
+    fi
+    if [ -d $SYSEXT_DIR ]; then
+        rm -rf "$SYSEXT_DIR"
+    fi
+
+    cp -r tmp/sysext "$SYSEXT_DIR"
+    systemd-sysext merge --force
+
+    echo
+    systemd-sysext status --no-pager
+
+    systemctl start cockpit.socket
+    exit 0
+fi
+
+. /run/host/etc/os-release
+if [ "$ID" != "fedora" ]; then
+    echo "systemd-sysext is currently only supported for Fedora hosts"
+    exit 1
+fi
+
+# build mode
+rm -rf tmp/sysext
+if [ ! -e Makefile ]; then
+    ./autogen.sh --prefix=/usr --sysconfdir=/etc --localstatedir=/var --enable-strict --enable-debug --disable-dependency-tracking
+fi
+make install DESTDIR=tmp/sysext
+install -D -m 644 tools/cockpit.pam tmp/sysext/usr/lib/pam.d/cockpit
```

Comment:
So this is one "Fedora" specific part, the arch <-> Fedora file is different. And ofcourse:

```
--libexecdir=/usr/lib/$pkgname/
```

But rumour has it that FHS wants to make `libexec` [mandatory](https://gitlab.freedesktop.org/FHS/fhs-spec/-/issues/6)

Reply:
But I believe /usr/libexec vs. /usr/lib/cockpit shouldn't matter much for the sysext -- that's mostly stylistic, and as long as you actually install the binaries to where they are expected (which should be the case) it should be alright.

Reply:
But indeed the PAM file needs to match. We can select Fedora, Debian, or Arch by os-release `$ID`, and fail for others?

Reply:
Reworked to support Debian/Ubuntu/Fedora/Centos/RHEL/Arch PAM files.